lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -1328,9 +1328,22 @@ static bool isClangTypeMoreIndirectThanSubstType(TypeConverter &TC,
   // Pass C++ const reference types indirectly. Right now there's no way to
   // express immutable borrowed params, so we have to have this hack.
   // Eventually, we should just express these correctly: rdar://89647503
-  if (clangTy->isReferenceType() &&
-      clangTy->getPointeeType().isConstQualified())
-    return true;
+    
+    /*
+     Why does this not compile?
+     if (clangTy->getPointeeType().isConstQualified() ||
+         clangTy->isRValueReferenceType()) {
+         
+         return true;
+     }
+     */
+    if (clangTy->isReferenceType()) {
```

Comment:
I don't think this is correct. Currently all `Cxx/Interop/stdlib` tests are passing, and currently still investigating the sil gen here